plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/utils/KGraphMappingUtil.xtend

```diff
@@ -126,9 +128,25 @@ class KGraphMappingUtil {
         
         skNode.position = new Point(kNode.xpos + leftInset, kNode.ypos + topInset)
         skNode.size = new Dimension(kNode.width, kNode.height)
-        for (property : KlighdDataManager.instance.preservedProperties) {
-            skNode.properties.put(property.id.substring(property.id.lastIndexOf('.') + 1), kNode.getProperty(property))
+
+        var properties = kNode.allProperties;
+        var propertyKeys = properties.keySet;
+        var blackList = KlighdDataManager.instance.blacklistedProperties;
+        for (key : propertyKeys) {
```

Comment:
For better efficiency here you could just run through the properties instead of only the keys. Removes the need to search through the properties via `get(key)` for every key